virtcontainers/factory/factory.go

```diff
@@ -100,6 +100,71 @@ func resetHypervisorConfig(config *vc.VMConfig) {
 	config.ProxyConfig = vc.ProxyConfig{}
 }
 
+func compareStruct(foo, bar reflect.Value) bool {
+	for i := 0; i < foo.NumField(); i++ {
```

Comment:
Missing check on field counts for `foo` and `bar`? They might compare as the same but if `bar` has extra fields that would be incorrect surely?

Reply:
They have to pass `foo.Type() == bar.Type()` so they must be of the same struct type.

Reply:
Ah - good point!